server/public/model/channel_bookmark.go

```diff
@@ -47,11 +47,68 @@ func (o *ChannelBookmark) Auditable() map[string]interface{} {
 	}
 }
 
+type ChannelBookmarkPatch struct {
+	FileId      *string              `json:"file_id"`
+	DisplayName *string              `json:"display_name"`
+	SortOrder   *int64               `json:"sort_order"`
+	LinkUrl     *string              `json:"link_url,omitempty"`
+	ImageUrl    *string              `json:"image_url,omitempty"`
+	Emoji       *string              `json:"emoji,omitempty"`
+	Type        *ChannelBookmarkType `json:"type"`
+}
+
+func (o *ChannelBookmarkPatch) Auditable() map[string]interface{} {
+	return map[string]interface{}{
+		"file_id": o.FileId,
+		"type":    o.Type,
+	}
+}
+
+func (o *ChannelBookmark) Patch(patch *ChannelBookmarkPatch) {
+	if patch.FileId != nil {
+		o.FileId = *patch.FileId
+	}
+
+	if patch.DisplayName != nil {
```

Comment:
should also be different that empty, no? same with the others or I'm missing something

Reply:
I don't think so. The rationale here is that if the field is present in the patch, we accept its contents, so if we're receiving a `"display_name": ""` that means that the user wants to set its value to empty, even if that's going to cause a validation error down the line